cmd/kubelet/app/options/options.go

```diff
@@ -34,9 +34,10 @@ import (
 type KubeletServer struct {
 	componentconfig.KubeletConfiguration
 
-	AuthPath      util.StringFlag // Deprecated -- use KubeConfig instead
-	KubeConfig    util.StringFlag
-	APIServerList []string
+	AuthPath          util.StringFlag // Deprecated -- use KubeConfig instead
+	KubeConfig        util.StringFlag
+	WaitForKubeConfig bool
```

Comment:
> I'm confused about what this is doing. Why wouldn't whatever is launching this wait until it had the information it needed.

Per #30360, we need to be able to start kubelet in the same way on masters and nodes, i.e. install it via a package manager and launch it with system init daemon, it there by will be started with the same flags and we need the ability point it at the API server after it's launched.

> If its managed in this file, why wouldn't we build an automatic interlock to avoid having an additional parameter?

Could you please explain this a little bit more? As I said, flag is being added to avoid breaking backwards compatibility... I'm happy to explore alternatives.